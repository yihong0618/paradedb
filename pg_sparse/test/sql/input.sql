SELECT '[1,2,3]'::svector;
SELECT '[-1,-2,-3]'::svector;
SELECT '[1.,2.,3.]'::svector;
SELECT ' [ 1,  2 ,    3  ] '::svector;
SELECT '[1.23456]'::svector;
SELECT '[hello,1]'::svector;
SELECT '[NaN,1]'::svector;
SELECT '[Infinity,1]'::svector;
SELECT '[-Infinity,1]'::svector;
SELECT '[1.5e38,-1.5e38]'::svector;
SELECT '[1.5e+38,-1.5e+38]'::svector;
SELECT '[1.5e-38,-1.5e-38]'::svector;
SELECT '[4e38,1]'::svector;
SELECT '[1,2,3'::svector;
SELECT '[1,2,3]9'::svector;
SELECT '1,2,3'::svector;
SELECT ''::svector;
SELECT '['::svector;
SELECT '[,'::svector;
SELECT '[]'::svector;
SELECT '[1,]'::svector;
SELECT '[1a]'::svector;
SELECT '[1,,3]'::svector;
SELECT '[1, ,3]'::svector;
SELECT '[1,2,3]'::svector(2);

SELECT unnest('{"[1,2,3]", "[4,5,6]"}'::svector[]);
SELECT '{"[1,2,3]"}'::svector(2)[];
