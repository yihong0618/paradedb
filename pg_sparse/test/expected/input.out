SELECT '[1,2,3]'::svector;
 svector 
---------
 [1,2,3]
(1 row)

SELECT '[-1,-2,-3]'::svector;
  svector   
------------
 [-1,-2,-3]
(1 row)

SELECT '[1.,2.,3.]'::svector;
 svector 
---------
 [1,2,3]
(1 row)

SELECT ' [ 1,  2 ,    3  ] '::svector;
 svector 
---------
 [1,2,3]
(1 row)

SELECT '[1.23456]'::svector;
  svector  
-----------
 [1.23456]
(1 row)

SELECT '[hello,1]'::svector;
ERROR:  invalid input syntax for type svector: "[hello,1]"
LINE 1: SELECT '[hello,1]'::svector;
               ^
SELECT '[NaN,1]'::svector;
ERROR:  NaN not allowed in svector
LINE 1: SELECT '[NaN,1]'::svector;
               ^
SELECT '[Infinity,1]'::svector;
ERROR:  infinite value not allowed in svector
LINE 1: SELECT '[Infinity,1]'::svector;
               ^
SELECT '[-Infinity,1]'::svector;
ERROR:  infinite value not allowed in svector
LINE 1: SELECT '[-Infinity,1]'::svector;
               ^
SELECT '[1.5e38,-1.5e38]'::svector;
      svector       
--------------------
 [1.5e+38,-1.5e+38]
(1 row)

SELECT '[1.5e+38,-1.5e+38]'::svector;
      svector       
--------------------
 [1.5e+38,-1.5e+38]
(1 row)

SELECT '[1.5e-38,-1.5e-38]'::svector;
      svector       
--------------------
 [1.5e-38,-1.5e-38]
(1 row)

SELECT '[4e38,1]'::svector;
ERROR:  infinite value not allowed in svector
LINE 1: SELECT '[4e38,1]'::svector;
               ^
SELECT '[1,2,3'::svector;
ERROR:  malformed svector literal: "[1,2,3"
LINE 1: SELECT '[1,2,3'::svector;
               ^
DETAIL:  Unexpected end of input.
SELECT '[1,2,3]9'::svector;
ERROR:  malformed svector literal: "[1,2,3]9"
LINE 1: SELECT '[1,2,3]9'::svector;
               ^
DETAIL:  Junk after closing right brace.
SELECT '1,2,3'::svector;
ERROR:  malformed svector literal: "1,2,3"
LINE 1: SELECT '1,2,3'::svector;
               ^
DETAIL:  Vector contents must start with "[".
SELECT ''::svector;
ERROR:  malformed svector literal: ""
LINE 1: SELECT ''::svector;
               ^
DETAIL:  Vector contents must start with "[".
SELECT '['::svector;
ERROR:  malformed svector literal: "["
LINE 1: SELECT '['::svector;
               ^
DETAIL:  Unexpected end of input.
SELECT '[,'::svector;
ERROR:  malformed svector literal: "[,"
LINE 1: SELECT '[,'::svector;
               ^
DETAIL:  Unexpected end of input.
SELECT '[]'::svector;
ERROR:  svector must have at least 1 dimension
LINE 1: SELECT '[]'::svector;
               ^
SELECT '[1,]'::svector;
ERROR:  invalid input syntax for type svector: "[1,]"
LINE 1: SELECT '[1,]'::svector;
               ^
SELECT '[1a]'::svector;
ERROR:  invalid input syntax for type svector: "[1a]"
LINE 1: SELECT '[1a]'::svector;
               ^
SELECT '[1,,3]'::svector;
ERROR:  malformed svector literal: "[1,,3]"
LINE 1: SELECT '[1,,3]'::svector;
               ^
SELECT '[1, ,3]'::svector;
ERROR:  invalid input syntax for type svector: "[1, ,3]"
LINE 1: SELECT '[1, ,3]'::svector;
               ^
SELECT '[1,2,3]'::svector(2);
ERROR:  expected 2 dimensions, not 3
SELECT unnest('{"[1,2,3]", "[4,5,6]"}'::svector[]);
 unnest  
---------
 [1,2,3]
 [4,5,6]
(2 rows)

SELECT '{"[1,2,3]"}'::svector(2)[];
ERROR:  expected 2 dimensions, not 3
